### PR TITLE
WT-2023 Avoid bit shifts during read-write lock operations.

### PR DESCRIPTION
**Pull request by michaelcahill**
I hope this doesn't muddy the waters, but I've wondered this before.  Is there any reason not to avoid the shift operations entirely?

### CODE REVIEW

**Comment by michaelcahill** [comment]
@keithbostic, @ddanderson can you please both take a look?

File: src/os_posix/os_mtx_rw.c [hunk 1]
```diff
@@ -63,7 +63,7 @@
  * next number available. However, instead of waiting for 'writers' to equal
  * their number, they wait for 'readers' to equal their number.
  *
- * This has the effect of queueing lock requests in the order they arrive
+ * This has the effect of queuing lock requests in the order they arrive
```
Comment (michaelcahill) [inline src/os_posix/os_mtx_rw.c]: this is according to `aspell` on my system...

**Comment by michaelcahill** [comment]
p.s. this should also make the code endian-independent.

File: src/os_posix/os_mtx_rw.c [hunk 2]
```diff
@@ -159,22 +156,17 @@ __wt_try_readlock(WT_SESSION_IMPL *session, WT_RWLOCK *rwlock)
...
 	/*
 	 * The old and new values for the lock: note the use of "users" instead
-	 * of "writers", this may not be the lock's current value, rather it's
-	 * the value the lock must have if we are to grant this write lock.
-	 *
-	 * Note the masking of "users + 1" in the calculation of the new value:
-	 * we want to set the readers field to the next readers value, not the
-	 * next users value, so it has to wrap, not overflow.
+	 * of "readers", this may not be the lock's current value, rather it's
+	 * the value the lock must have if we are to grant this read lock.
```
Comment (michaelcahill) [inline src/os_posix/os_mtx_rw.c]: I think we should make this comment change regardless of the code changes: there was a copy/paste swap of "write" for "read" that I didn't spot on my review.

**Comment by keithbostic** [comment]
@michaelcahill, @ddanderson, this looks correct to me, and I think it's worth the price of admission just remove the endian dependency for read-write locks.

I can't imagine it's slower either, the compiler should be generating the same masking instructions that we were doing by hand. (And, we may do less instructions because the fast test from c76a969 means the old value is correct, so there's no need to touch it, we just have to create the new value.)

**Pull request merged**